### ISSUE TITLE
Achieve 50 TPS or 300 txs per a block in test harness

### ISSUE THREAD

**Issue by goshawk-3** [dusk-network/dusk-blockchain#997]
Currently, test harness reports up to 10 TPS which in most cases means  around 56 transactions accepted within 5-6 block time.

Obviously, the main limitation comes from two constants we've got:

```go

// MaxTxSetSize defines the maximum amount of transactions.
// It is TBD along with block size and processing.MaxFrameSize.
const MaxTxSetSize = 150000

// MaxFrameSize is set at 250000 bytes.
MaxFrameSize = uint64(250000)

```

**Comment by goshawk-3** [comment]
Highest result achieved so far.
```
	 msg="Measured TPS 11.20 (56/5)" height=14
	 msg="Measured TPS 9.33 (56/6)" height=15
	 msg="Measured TPS 9.33 (56/6)" height=15
	 msg="Measured TPS 9.33 (56/6)" height=16
	 msg="Measured TPS 11.20 (56/5)" height=17
	 msg="Measured TPS 11.20 (56/5)" height=17
	 msg="Measured TPS 9.33 (56/6)" height=18
	 msg="Measured TPS 9.33 (56/6)" height=19
	 msg="Measured TPS 9.33 (56/6)" height=19
	 msg="Measured TPS 9.33 (56/6)" height=19
	 msg="Measured TPS 9.33 (56/6)" height=19
```

**Comment by julesdesmit** [comment]
After some investigation and quick fixes, we managed to get to this result:

```
time="2021-03-31T20:15:33+02:00" level=info msg="Measured TPS 3.80 (19/5) 19" cumulative_tx_count=19 height=7
time="2021-03-31T20:15:37+02:00" level=info msg="Measured TPS 29.20 (146/5) 165" cumulative_tx_count=165 height=8
time="2021-03-31T20:15:45+02:00" level=info msg="Measured TPS 61.20 (306/5) 471" cumulative_tx_count=471 height=9
time="2021-03-31T20:15:49+02:00" level=info msg="Measured TPS 51.00 (306/6) 777" cumulative_tx_count=777 height=10
time="2021-03-31T20:15:53+02:00" level=info msg="Measured TPS 61.20 (306/5) 1083" cumulative_tx_count=1083 height=11
time="2021-03-31T20:16:01+02:00" level=info msg="Measured TPS 51.00 (306/6) 1389" cumulative_tx_count=1389 height=12
time="2021-03-31T20:16:05+02:00" level=info msg="Measured TPS 61.20 (306/5) 1695" cumulative_tx_count=1695 height=13
time="2021-03-31T20:16:09+02:00" level=info msg="Measured TPS 61.20 (306/5) 2001" cumulative_tx_count=2001 height=14
time="2021-03-31T20:16:17+02:00" level=info msg="Measured TPS 51.00 (306/6) 2307" cumulative_tx_count=2307 height=15
^[2^[1time="2021-03-31T20:16:21+02:00" level=info msg="Measured TPS 61.20 (306/5) 2613" cumulative_tx_count=2613 height=16
time="2021-03-31T20:16:25+02:00" level=info msg="Measured TPS 51.00 (306/6) 2919" cumulative_tx_count=2919 height=17
time="2021-03-31T20:16:33+02:00" level=info msg="Measured TPS 61.20 (306/5) 3225" cumulative_tx_count=3225 height=18
time="2021-03-31T20:16:37+02:00" level=info msg="Measured TPS 51.00 (306/6) 3531" cumulative_tx_count=3531 height=19
time="2021-03-31T20:16:41+02:00" level=info msg="Measured TPS 61.20 (306/5) 3837" cumulative_tx_count=3837 height=20
time="2021-03-31T20:16:49+02:00" level=info msg="Measured TPS 61.20 (306/5) 4143" cumulative_tx_count=4143 height=21
time="2021-03-31T20:16:53+02:00" level=info msg="Measured TPS 51.00 (306/6) 4449" cumulative_tx_count=4449 height=22
time="2021-03-31T20:17:02+02:00" level=info msg="Measured TPS 61.20 (306/5) 4755" cumulative_tx_count=4755 height=23
time="2021-03-31T20:17:06+02:00" level=info msg="Measured TPS 61.20 (306/5) 5061" cumulative_tx_count=5061 height=24
time="2021-03-31T20:17:10+02:00" level=info msg="Measured TPS 51.00 (306/6) 5367" cumulative_tx_count=5367 height=25
time="2021-03-31T20:17:18+02:00" level=info msg="Measured TPS 61.20 (306/5) 5673" cumulative_tx_count=5673 height=26
time="2021-03-31T20:17:22+02:00" level=info msg="Measured TPS 51.00 (306/6) 5979" cumulative_tx_count=5979 height=27
time="2021-03-31T20:17:26+02:00" level=info msg="Measured TPS 61.20 (306/5) 6285" cumulative_tx_count=6285 height=28
time="2021-03-31T20:17:34+02:00" level=info msg="Measured TPS 51.00 (306/6) 6591" cumulative_tx_count=6591 height=29
time="2021-03-31T20:17:38+02:00" level=info msg="Measured TPS 61.20 (306/5) 6897" cumulative_tx_count=6897 height=30
time="2021-03-31T20:17:42+02:00" level=info msg="Measured TPS 51.00 (306/6) 7203" cumulative_tx_count=7203 height=31
time="2021-03-31T20:17:50+02:00" level=info msg="Measured TPS 61.20 (306/5) 7509" cumulative_tx_count=7509 height=32
time="2021-03-31T20:17:54+02:00" level=info msg="Measured TPS 51.00 (306/6) 7815" cumulative_tx_count=7815 height=33
time="2021-03-31T20:17:58+02:00" level=info msg="Measured TPS 61.20 (306/5) 8121" cumulative_tx_count=8121 height=34
time="2021-03-31T20:18:06+02:00" level=info msg="Measured TPS 61.20 (306/5) 8427" cumulative_tx_count=8427 height=35
time="2021-03-31T20:18:10+02:00" level=info msg="Measured TPS 51.00 (306/6) 8733" cumulative_tx_count=8733 height=36
time="2021-03-31T20:18:14+02:00" level=info msg="Measured TPS 61.20 (306/5) 9039" cumulative_tx_count=9039 height=37
time="2021-03-31T20:18:22+02:00" level=info msg="Measured TPS 2.20 (11/5) 9050" cumulative_tx_count=9050 height=38
time="2021-03-31T20:18:26+02:00" level=info msg="Measured TPS 0.20 (1/5) 9051" cumulative_tx_count=9051 height=39
```

Merging a fix for #1003 will close this issue.
